lib/AST/ASTPrinter.cpp

```diff
@@ -1291,31 +1308,36 @@ void PrintAST::printPattern(const Pattern *pattern) {
 
   case PatternKind::EnumElement: {
     auto elt = cast<EnumElementPattern>(pattern);
-    // FIXME: Print element expr.
+    Printer << "." << elt->getElementDecl()->getBaseName();
     if (elt->hasSubPattern())
       printPattern(elt->getSubPattern());
     break;
   }
 
   case PatternKind::OptionalSome:
     printPattern(cast<OptionalSomePattern>(pattern)->getSubPattern());
-    Printer << '?';
```

Comment:
Why remove this? Do we still print the `?` for e.g:

```
func foo(_ x: Int?) {
  switch x {
  case let x?:
    break
  case nil:
    break
  }
}
```

Reply:
Fixed and added test.